src/fable/Fable.Compiler/Replacements.fs

```diff
@@ -1226,7 +1226,11 @@ module private AstPass =
         | "removeAt" ->
             icall "splice" (c.Value, [args.Head; makeConst 1]) |> Some
         | "reverse" when kind = Array ->
-            icall "reverse" (instanceArgs c i.args) |> Some
+            match i.returnType with
+            | Fable.Array _ ->
+                emit i "$0.slice().reverse()" i.args |> Some
```

Comment:
Ah, ok! I finally got it, sorry 😅 So for `ResizeArray` this is mutating but not for for `Array.rev`. Maybe we could add a comment explaining this to avoid some head scratching to my future me? ;)